### PR TITLE
Release 3.18.0

### PR DESCRIPTION
**Pull request by jennyf19**
Release 3.17.4 w/GDPR & sovereign cloud

### CODE REVIEW

**Comment by henrik-me** [comment]
looks great. codeflow only handles 64 commits... I heard Jean-Marc is testing this one :)

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationResult.cs [hunk 1]
```diff
@@ -122,9 +122,9 @@ internal AuthenticationResult(string accessTokenType, string accessToken, DateTi
         public string IdToken { get; internal set; }
 
         /// <summary>
-        /// Gets the authority which has issued tokens.
+        /// Gets the authority that has issued the token.
         /// </summary>
-        internal string Authority { get; set; }
```
Comment (jmprieur) [inline src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationResult.cs]: This is an addition to the public API. We need to increase the minor number of the release (3.18.0, not 3.17.4)

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/IAdalLogCallback.cs [hunk 2]
```diff
@@ -57,8 +58,20 @@ public enum LogLevel
...
 
     /// <summary>
-    /// Callback for capturing ADAL logs to custom logging schemes.
+    /// Callback delegate that allows the developer to consume logs handle them in a custom manner.
     /// </summary>
+    /// <param name="level">Log level of the message</param>
+    /// <param name="message">Pre-formatted log message</param>
+    /// <param name="containsPii">Indicates if the log message contains PII. If Logger.PiiLoggingEnabled is set to 
+    /// false then this value is always false.</param>
+    public delegate void LogCallback(LogLevel level, string message, bool containsPii);
```
Comment (jmprieur) [inline src/Microsoft.IdentityModel.Clients.ActiveDirectory/IAdalLogCallback.cs]: This is an addition to the public API (and therefore we need to increase the release minor number) => 3.18 and not 3.17.4

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/IAdalLogCallback.cs [hunk 3]
```diff
@@ -76,18 +89,28 @@ public sealed class LoggerCallbackHandler
     {
         private static readonly object LockObj = new object();
 
-        private static IAdalLogCallback _localCallback;
+        /// <summary>
+        /// Flag to enable/disable logging of PII data. PII logs are never written to default outputs like Console, Logcat or NSLog.
+        /// Default is set to false.
+        /// </summary>
+        public static bool PiiLoggingEnabled { get; set; } = false;
```
Comment (jmprieur) [inline src/Microsoft.IdentityModel.Clients.ActiveDirectory/IAdalLogCallback.cs]: This is an addition to the public API (and therefore we need to increase the release minor number) => 3.18 and not 3.17.4

**Review by jmprieur — CHANGES_REQUESTED** [review]
As explained in the details, there is a breaking change in the public API, which we can fix (as this is a change of namspace)

There are also, as expected additional changes to the public API, and therefore we need to increase the release number (3.18 and not 3.17.4)

The rest looks good to be, althought with that many files I might have missed issues (but for sure not in the public API :))

File: changelog.txt [hunk 4]
```diff
@@ -1,3 +1,10 @@
+Version 3.17.4
```
Comment (jmprieur) [inline changelog.txt]: Should be 3.18, really as we have additions to the public API.

**Comment by jmprieur** [comment]
We need to update the README to explain the new Logging mechanism. Proposed PR #955 for this.

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationResult.cs (hunk 1 above)
Comment (jennyf19) [reply src/Microsoft.IdentityModel.Clients.ActiveDirectory/AuthenticationResult.cs]: I increased the minor number to 3.18.0

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/IAdalLogCallback.cs (hunk 2 above)
Comment (jennyf19) [reply src/Microsoft.IdentityModel.Clients.ActiveDirectory/IAdalLogCallback.cs]: done.

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/IAdalLogCallback.cs (hunk 3 above)
Comment (jennyf19) [reply src/Microsoft.IdentityModel.Clients.ActiveDirectory/IAdalLogCallback.cs]: done.

File: src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/net45/CustomWebBrowser.cs [hunk 5]
```diff
@@ -30,7 +30,7 @@
 using System.Runtime.InteropServices;
 using System.Windows.Forms;
 
-namespace Microsoft.IdentityModel.Clients.ActiveDirectory.Internal
```
[the remark this answers is not in the document]
Comment (jennyf19) [reply src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/net45/CustomWebBrowser.cs]: Got it. Thanks

File: changelog.txt (hunk 4 above)
Comment (jennyf19) [reply changelog.txt]: done.

**Review by jmprieur — APPROVED** [review]

**Pull request merged**